crates/cast/src/starknet_commands/account/create.rs

```diff
@@ -162,24 +153,31 @@ pub fn add_created_profile_to_configuration(
         )
         .unwrap();
 
-    let package = &metadata.packages[0].manifest_metadata.tool;
-
-    let property = get_property(package, &Some(name.clone()), "account");
-    if property.is_ok() {
-        return Err(anyhow!(
-            "Failed to add {name} profile to the Scarb.toml. Profile already exists"
-        ));
+    if let Ok(tool_sncast) = get_package_tool_sncast(&metadata) {
+        let property = tool_sncast
+            .get(&config.account)
+            .and_then(|profile_| profile_.get("account"));
+        if property.is_some() {
+            bail!(
+                "Failed to add {} profile to the Scarb.toml. Profile already exists",
+                config.account
+            );
+        }
     }
 
     let toml_string = {
         let mut tool_sncast = toml::value::Table::new();
         let mut new_profile = toml::value::Table::new();
 
-        new_profile.insert("network".to_string(), Value::String(network));
-        new_profile.insert("url".to_string(), Value::String(url));
-        new_profile.insert("account".to_string(), Value::String(name.clone()));
+        new_profile.insert("network".to_string(), Value::String(config.network.clone()));
+        new_profile.insert("url".to_string(), Value::String(config.rpc_url.clone()));
+        new_profile.insert("account".to_string(), Value::String(config.account.clone()));
+        new_profile.insert(
```

Comment:
Should this always be inserted, even if it's pointing to the default file?

Reply:
yes